### PR TITLE
packages: add RHEL 9 support

### PR DESCRIPTION
**Pull request by defanator**

### CODE REVIEW

File: packages/nginx-amplify-agent/rpm/nginx-amplify-agent.spec [hunk 1]
```diff
@@ -23,13 +23,22 @@ BuildRequires: python3-pip
 %if 0%{?amzn} >= 2
 Requires: python3 >= 3.7
 Requires: python3-requests
-%else
```
Comment (amudukutore) [inline packages/nginx-amplify-agent/rpm/nginx-amplify-agent.spec:26]: Looks like the `else` block has been removed and 2 new checks for rhel8 and rhel9 have been added. Are there any code paths that can take the `else` clause that are not `rhel` ? If so, none of the Python Requires logic will take effect for those use caseS.

**Comment by amudukutore** [comment]
@defanator - looks good to me. Just had one minor question. Have you tested the new changes to make sure that they work for `rhel8` and `rhel9` as well as one non-rhel use case?

**Review by beckelhimerf5 — APPROVED** [review]

File: packages/nginx-amplify-agent/rpm/nginx-amplify-agent.spec (hunk 1 above)
Comment (defanator) [reply packages/nginx-amplify-agent/rpm/nginx-amplify-agent.spec:26]: We currently list RHEL 8 and Amazon Linux as supported rpm-based distros only, so no reason to worry about other ones (e.g. opensuse/SLES or Fedora).

In theory current spec should work flawlessly on RHEL derivatives such as AlmaLinux and RockyLinux, but as these are binary compatible with RHEL, we don't need to worry either.

**Comment by defanator** [comment]
@amudukutore absolutely, this actually is a part of our CI stuff in internal repository. With this change, agent can be built on RHEL 8, RHEL 9, and Amazon Linux.

**Pull request merged**